content/snippets/easy_to_change.md

```diff
@@ -0,0 +1,32 @@
+title: Easy to Test, Easy To Change
+tags: software engineering
+author: cdent
+created: 2024-02-23
+
+In conversation with my wife recently I mentioned that I've been doing software
+development for more than 40 years. I've learned, forgotten, and re-learned
+many things in that time.
+
+There are two goals I now accept as true and self evident but are neither
+readily nor universally implemented, especially in high pressure product
+development environments where short-term progress is traded against long-term
+success.
+
+Those two goals are more important than "providing the functionality we want".
+They are that code must be:
+
+1. easy to change
+2. easy to test
+
+Without these, maintaining the functionality we want (as that functionality
+evolves) becomes far too hard.
```

Comment:
```suggestion
Without these, maintaining the functionality we want becomes far too hard -- especially as that functionality evolves.
```

not entirely sure about this one

Reply:
I feel like it changes the emphasis, so reject this one

Reply:
makes sense